src/main/java/graphql/nadel/result/ResultComplexityAggregator.java

```diff
@@ -0,0 +1,46 @@
+package graphql.nadel.result;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class ResultComplexityAggregator {
+    private Integer totalNodeCount;
+    private LinkedHashMap<String, Integer> serviceNodeCountsMap;
+
+
+    public ResultComplexityAggregator(Integer totalNodeCount, LinkedHashMap<String, Integer> serviceNodeCountsMap) {
+        this.totalNodeCount = totalNodeCount;
+        this.serviceNodeCountsMap = serviceNodeCountsMap;
+    }
+
+    public Integer getTotalNodeCount() {
+        return totalNodeCount;
+    }
+
+    public LinkedHashMap<String, Integer> getServiceNodeCountsMap() {
+        return serviceNodeCountsMap;
+    }
+
+    public void addAndSetServiceNodeCount(String serviceFieldName, Integer nodeCount) {
+        serviceNodeCountsMap.putIfAbsent(serviceFieldName, 0);
+        serviceNodeCountsMap.put(serviceFieldName, serviceNodeCountsMap.get(serviceFieldName) + nodeCount);
+        totalNodeCount += nodeCount;
```

Comment:
there is a method on maps called compute

https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#compute-K-java.util.function.BiFunction-

`map.compute(serviceFieldName, (k, v) -> (v == null) ? 0 : v+serviceCount)`

Its basically what you have there but in a more compact way